array.chain.js

```diff
@@ -74,4 +77,9 @@ class AsyncArray {
 
 }
 
-module.exports = AsyncArray;
+metasync.for = (
+  // Create an AsyncArray instance
```

Comment:
@aqrln Fixed. BTW, I renamed `AsyncArray` to `ArrayChain` because it implements chaining syntax but becomes asynс only if callback contains I/O or timers. In other cases it would be great to use it even instead of `forEach` in applied code to make sync and async array iterations consistent.